### PR TITLE
[CELEBORN-1600] Enable check server dependencies

### PR DESCRIPTION
**Pull request by cxzl25**
### Why are the changes needed?
Server module missing checks.

### Does this PR introduce _any_ user-facing change?
No

### How was this patch tested?
GA

### CODE REVIEW

**Comment by FMX** [comment]
Thanks, it would be helpful to connect this PR to a Jira ticket.

File: project/CelebornBuild.scala [hunk 1]
```diff
@@ -185,7 +188,7 @@ object Dependencies {
   val httpClient5 = "org.apache.httpcomponents.client5" % "httpclient5" % httpClient5Version
   val httpCore5 = "org.apache.httpcomponents.core5" % "httpcore5" % httpCore5Version
   val httpCore5H2 = "org.apache.httpcomponents.core5" % "httpcore5-h2" % httpCore5Version
-  val javaxAnnotationApi = "javax.annotation" % "javax.annotation-api" % javaxAnnotationApiVersion
+  val jakartaAnnotationApi = "jakarta.annotation" % "jakarta.annotation-api" % jakartaAnnotationApiVersion
```
Comment (cxzl25) [inline project/CelebornBuild.scala:191]: Introduced by CELEBORN-1477, the dependencies of SBT and maven are different.

File: dev/deps/dependencies-client-mr [hunk 2]
```diff
@@ -72,13 +72,13 @@ jackson-annotations/2.15.3//jackson-annotations-2.15.3.jar
 jackson-core-asl/1.9.13//jackson-core-asl-1.9.13.jar
 jackson-core/2.15.3//jackson-core-2.15.3.jar
 jackson-databind/2.15.3//jackson-databind-2.15.3.jar
-jackson-jaxrs-base/2.12.7//jackson-jaxrs-base-2.12.7.jar
-jackson-jaxrs-json-provider/2.12.7//jackson-jaxrs-json-provider-2.12.7.jar
+jackson-jaxrs-base/2.15.3//jackson-jaxrs-base-2.15.3.jar
```
Comment (cxzl25) [inline dev/deps/dependencies-client-mr:75]: ### main
`jackson-jaxrs-base-2.13.2.jar`  `jackson-jaxrs-base-2.15.3.jar`

```
ls -al | grep jackson
-rw-r--r--@   1 csy  wheel     75565  9 18 18:33 jackson-annotations-2.15.3.jar
-rw-r--r--@   1 csy  wheel    549233  9 18 18:33 jackson-core-2.15.3.jar
-rw-r--r--@   1 csy  wheel   1620120  9 18 18:33 jackson-databind-2.15.3.jar
-rw-r--r--@   1 csy  wheel     19662  9 18 18:33 jackson-databind-nullable-0.2.6.jar
-rw-r--r--@   1 csy  wheel     54629  9 18 18:33 jackson-dataformat-yaml-2.15.3.jar
-rw-r--r--@   1 csy  wheel    122937  9 18 18:33 jackson-datatype-jsr310-2.15.3.jar
-rw-r--r--@   1 csy  wheel     31149  9 18 18:33 jackson-jaxrs-base-2.13.2.jar
-rw-r--r--@   1 csy  wheel     31857  9 18 18:33 jackson-jaxrs-base-2.15.3.jar
-rw-r--r--@   1 csy  wheel     15964  9 18 18:33 jackson-jaxrs-json-provider-2.13.2.jar
-rw-r--r--@   1 csy  wheel     16636  9 18 18:33 jackson-jaxrs-json-provider-2.15.3.jar
-rw-r--r--@   1 csy  wheel     36826  9 18 18:33 jackson-module-jaxb-annotations-2.14.1.jar
-rw-r--r--@   1 csy  wheel     36668  9 18 18:33 jackson-module-jaxb-annotations-2.15.3.jar
-rw-r--r--@   1 csy  wheel    513956  9 18 18:33 jackson-module-scala_2.12-2.15.3.jar
-rw-r--r--@   1 csy  wheel     76924  9 18 18:33 jersey-media-json-jackson-2.39.1.jar
```


### PR

```
-rw-r--r--@   1 csy  wheel     75565  9 18 18:54 jackson-annotations-2.15.3.jar
-rw-r--r--@   1 csy  wheel    549233  9 18 18:54 jackson-core-2.15.3.jar
-rw-r--r--@   1 csy  wheel   1620120  9 18 18:54 jackson-databind-2.15.3.jar
-rw-r--r--@   1 csy  wheel     19662  9 18 18:54 jackson-databind-nullable-0.2.6.jar
-rw-r--r--@   1 csy  wheel     54629  9 18 18:54 jackson-dataformat-yaml-2.15.3.jar
-rw-r--r--@   1 csy  wheel    122937  9 18 18:54 jackson-datatype-jsr310-2.15.3.jar
-rw-r--r--@   1 csy  wheel     31857  9 18 18:54 jackson-jaxrs-base-2.15.3.jar
-rw-r--r--@   1 csy  wheel     16636  9 18 18:54 jackson-jaxrs-json-provider-2.15.3.jar
-rw-r--r--@   1 csy  wheel     36668  9 18 18:54 jackson-module-jaxb-annotations-2.15.3.jar
-rw-r--r--@   1 csy  wheel    513956  9 18 18:54 jackson-module-scala_2.12-2.15.3.jar
-rw-r--r--@   1 csy  wheel     76924  9 18 18:54 jersey-media-json-jackson-2.39.1.jar
```

File: dev/dependencies.sh [hunk 3]
```diff
@@ -208,7 +206,7 @@ case "$MODULE" in
     ;;
   *)
     MODULE="server"
-    MVN_MODULES="worker,master"
+    MVN_MODULES="worker,master,cli"
```
Comment (cxzl25) [inline dev/dependencies.sh:209]: The cli project introduces picocli dependencies.

**Comment by cxzl25** [comment]
@pan3793 @cfmcgrady @waitinfuture @turboFei

File: dev/dependencies.sh [hunk 4]
```diff
@@ -56,13 +56,29 @@ function mvn_build_classpath() {
...
 
 function sbt_build_client_classpath() {
-  PATTERN="$SBT_PROJECT / Runtime / managedClasspath"
-  deps=$(
-    $SBT -P$MODULE "clean; export Runtime/managedClasspath" | \
-      awk -v pat="$PATTERN" '$0 ~ pat { found=1 } found { print }' | \
-      awk 'NR==2' | \
-      tr ":" "\n"
-  )
+    PATTERN="$SBT_PROJECT / Runtime / managedClasspath"
```
Comment (cfmcgrady) [inline dev/dependencies.sh:59]: indent

**Review by cfmcgrady — APPROVED** [review]
LGTM

**Review by turboFei — APPROVED** [review]

**Review by FMX — APPROVED** [review]
LGTM. Thanks. Merged into main(v0.6.0).

**Pull request closed without merging**